scripts/dev/roxdebug.sh

```diff
@@ -99,23 +97,26 @@ echo
 ensure_target_filesystem_is_writable() {
   local read_only_root_fs
   if [[ "${target_is_pod}" == "true" ]]; then
-    read_only_root_fs=$(kubectl -n stackrox get "${deployment}" \
-      -o=jsonpath="{.spec.containers[?(@.name=='$container_name')].securityContext.readOnlyRootFilesystem}")
+    read_only_root_fs="$(kubectl -n stackrox get "${deployment}" \
+      -o=jsonpath="{.spec.containers[?(@.name=='${container_name}')].securityContext.readOnlyRootFilesystem}")"
     if [[ $read_only_root_fs == "true" ]]; then
       echo >&2 "Cannot debug ${deployment}: the pod uses a read-only root filesystem"
       exit 1
     fi
   else
-    read_only_root_fs=$(kubectl -n stackrox get "${deployment}" \
-      -o=jsonpath="{.spec.template.spec.containers[?(@.name=='${container_name}')].securityContext.readOnlyRootFilesystem}")
+    read_only_root_fs=$(kubectl -n stackrox get "${deployment}" -o=jsonpath="{.spec.template.spec.containers[?(@.name=='${container_name}')].securityContext.readOnlyRootFilesystem}")
```

Comment:
This one is still unquoted, but that is optional. I think the chance of receiving here a string containing spaces is low.

Reply:
I can do that fort consistency, but, out of curiosity, is it actually required -- I mean, even assuming spaces to appear here. Shellcheck did not complain, and, according to my tests:

```
$ bash
$ foo=$(echo foo bar)
$ echo "'$foo'"
'foo bar'
$
```

Reply:
Indeed it is my rule of a thumb that is not always required. I am unable to give you a quick example currently that would show the dangers of not quoting. But I can share one when I find/recall one :)